### DISCUSSION TITLE
Emojis in the new Terminal Logger output

### DISCUSSION THREAD

**Discussion by bsdayo** [dotnet/msbuild#8768] [Ideas]
## Background

.NET 8 Preview 4 was released today and there are a lot of exciting features in it. First up is the new Terminal Logger of MSBuild.

I couldn't wait to install the new SDK and try out this new feature. It's very nice and meets my expectations for a modern compiler's output, beautiful and well organized. However, I noticed a somewhat annoying glitch: when the build fails, MSBuild outputs an red "❌" Emoji, like this (I left a compile error on purpose :D):

```
PS C:\Users\bsdayo\Code\TestTL> dotnet build /tl
MSBuild version 17.7.0-preview-23251-02+59879b095 for .NET
Restore complete (1.8s)
  TestTL failed with errors (10.3s)
    ❌︎ C:\Users\bsdayo\Code\TestTL\Program.cs(2,9): error CS0117: 'Console' does not contain a definition for 'HappyHappyHappy'
 ## ^ Here! ##

Build failed with errors in 14.7s
```

## Problems

The "❌" (or "⚠︎" in warnings) output here has, in my opinion, some drawbacks, as follows:

- One of the extremely important attractions of the terminal is that it is made of plain text (letters, characters only). For many people (including me), it is preferable to have a command line tool that also outputs plain text.
- In many terminals, the widths of Emoji and characters do not exactly correspond, even when using monospace fonts. This creates a sense of incongruity.
- The Emoji icons are vectorial (which can be seen as very high resolution), color-fixed and do not change as the terminal font, color, theme, etc. changes. For example, if I set the Terminal font to an monospace pixel font, but the Emoji is still a vector image, this also creates an incongruous feeling.

## The solution I envision

Simply remove the "❌" and "⚠︎". The whole error/warning message is in red or yellow, and it's already eye-catching.

That's my personal feedback on the new Terminal Logger. If you have ideas about this, feel free to discuss them below ;D

**Comment by fr1g** [comment]
Here's my opinion: add a switch for it, whether use emoji for an attractive view or a retro one🤣

**Reply by bsdayo** [reply]
If so, I would still prefer the Emoji display to be turned off by default XD

**Comment by bsdayo** [comment]
I also noticed that the sizes of these Emojis are not the same.

![Error Emoji](https://github.com/dotnet/msbuild/assets/41754841/4453a934-766f-484e-87fd-a9d067f28fba)
![Warning Emoji](https://github.com/dotnet/msbuild/assets/41754841/14df5766-f6ad-4684-bbe6-c71f8b2983d0)

Environment: Windows 11, Windows Terminal.

**Reply by bsdayo** [reply]
Created a new issue for this at #8769.

**Comment by JanKrivanek** [comment]
Emojis are not part of -tl output anymore (based on the feedback provided here) - so closing this discussion.
